### PR TITLE
Import export

### PR DESCRIPTION
**Pull request by igrosso**

### CODE REVIEW

File: client/components/Dashboard/icons/ImportExportIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { Component, PropTypes } from 'react';
```
Comment (sericaia) [inline client/components/Dashboard/icons/ImportExportIcon.jsx]: this should be no longer added to client/components/Dashboard/icons/ImportExportIcon.jsx, but client/components/Icons/ImportExportIcon.jsx... please rebase with dev branch..

File: client/routes.js [hunk 2]
```diff
@@ -25,7 +25,6 @@ export default (history) =>
       <Route path="users" component={containers.Users}>
         <Route path=":id" component={containers.User} />
       </Route>
-      <Route path="import-export" component={containers.ImportExport} />
```
Comment (sericaia) [inline client/routes.js]: we should still have a route to import/export inside configuration. We will need it here https://github.com/auth0/auth0-authorization-extension/blob/dev/client/actions/configuration.js#L30
and this line should be changed to '/configuration/import-export'

File: client/components/Configuration/ImportExport.jsx [hunk 3]
```diff
@@ -49,22 +40,46 @@ export default connectContainer(class extends Component {
...
+
   render() {
-    const { error, loading, record, children } = this.props;
+    const { error, loading, record, children, requesting, preview } = this.props;
     if (this.props.children) {
       return children;
     }
     return (
       <div>
+        <Confirm title='Are you sure?' show={ requesting } loading={ loading }
```
Comment (sericaia) [inline client/components/Configuration/ImportExport.jsx]: linting errors: it should be show={requesting} loading={loading} and so on

File: client/components/Configuration/ImportExport.jsx [hunk 4]
```diff
@@ -49,22 +40,46 @@ export default connectContainer(class extends Component {
     }
   }
 
+  confirmImport = () => {
+    const preview = this.props.preview;
+    this.props.importConfig(preview.toJSON())
+  }
+
+  closeImport = () => {
+    this.refs.file.value = '';
```
Comment (sericaia) [inline client/components/Configuration/ImportExport.jsx]: Refs are no longer encouraged: https://facebook.github.io/react/docs/refs-and-the-dom.html
do we really need to add them here? :S

**Comment by sericaia** [comment]
Hi!

Looking at this PR it doesn't seem that immutableJS structure was removed from last version. e.g field should not start with the following, but only show entries field.
```
{
  "__altered": false,
  "_root": {
    "ownerID": {},
(...)
```

Did you took this into account? Thanks :)

File: client/components/Configuration/ImportExport.jsx [hunk 5]
```diff
@@ -49,22 +40,46 @@ export default connectContainer(class extends Component {
...
+function mapStateToProps(state) {
+  return {
+    error: state.importExport.get('error'),
+    loading: state.importExport.get('loading'),
+    record: state.importExport.get('record'),
+    requesting: state.importExport.get('requesting'),
+    preview: state.importExport.get('preview')
+  }
+}
+export default connect(mapStateToProps, actions)(ImportExport);
```
Comment (sandrinodimattia) [inline client/components/Configuration/ImportExport.jsx]: Components should not be using connect. Should this be a container?

File: client/components/Configuration/RuleSettings.jsx [hunk 6]
```diff
@@ -4,6 +4,7 @@ import { Error, LoadingPanel, SectionHeader } from 'auth0-extension-ui';
 
 import RuleConfigurationTab from './RuleConfigurationTab';
 import APIAccessTab from './APIAccessTab';
+import ImportExport from './ImportExport';
```
Comment (sericaia) [inline client/components/Configuration/RuleSettings.jsx]: For consistency I suggest to change the name to ImportExportTab

File: client/containers/ConfigurationRule.jsx [hunk 7]
```diff
@@ -1,29 +1,32 @@
...
   static propTypes = {
     configuration: PropTypes.object,
-    fetchConfiguration: PropTypes.func.isRequired,
+    fetchConfiguration: PropTypes.func.required,
     saveConfiguration: PropTypes.func.isRequired,
-    fetchConfigurationResourceServer: PropTypes.func.isRequired,
+    fetchConfigurationResourceServer: PropTypes.func.required,
     saveConfigurationResourceServer: PropTypes.func.isRequired,
-    removeConfigurationResourceServer: PropTypes.func.isRequired
+    exportConfig: PropTypes.func.isRequired,
```
Comment (sericaia) [inline client/containers/ConfigurationRule.jsx]: remove extra comma

**Pull request merged**